data/macvtap/002-macvtap-daemonset.yaml

```diff
@@ -0,0 +1,63 @@
+---
+kind: ConfigMap
```

Comment:
Please try to edit this CM deployed by CNAO. I have a bad feeling the settings would get reverted by the operator. e.g. deploy macvtap, edit the CM, additionally deploy ovs, see if the CM remained in the original state.

If the CM changes, we have two options:
* Ask user to deploy it themselves
* Add explicit setting to CNAO not to touch it

I feel the former is better.

Reply:
Yup, after updating the value, the operator puts it back the way it was.

Remember that in https://github.com/kubevirt/macvtap-cni/pull/11 the configmap is still required, and that the macvtap DS won't come up until the config map is available.

That would require us (at least in the tests) to create the config map by some other means (hacky move).

How can we add the explicit setting to CNAO to stay away from the config map ? What are the disadvantages of this option ?

Reply:
Disadvantage is that we spoil the generic code with "if macvtap DP".

I was writing a guide on "how to hack it", but I think there is an elegant way too.

So this issue where we want to keep user changes on objects deployed is pretty common, although we were successful avoiding addressing it, until now.

So let's think how can we make it generic in the CNAO framework and keep everything specific to the component under data/ or pkg/network.

*Option A*: The simplest fit into the current API would be to introduce a Kubernetes object label `networkaddonsoperator.network.kubevirt.io/untouchable: "true"` (or any better name after `/`). That can be set by a user who wants to debug manual changes on any deployed component (e.g. they want to change resource requests, add debugging flags to running apps etc). It can be also set by default under data/ manifests. The only code we need to introduce then would live around https://github.com/kubevirt/cluster-network-addons-operator/blob/master/pkg/apply/apply.go#L37, where we check if an existing object we want to update has this flag, if it does, we ignore it. If you implement it, please do it in a separate PR (in case we need to backport it) and document it at the bottom of the README.

I am tempted to say "let's just keep it as it is and wait for the proper API", but that would make it harder to test under kubevirt and if somebody unaware uses it (despite it us "experimental", they may get hurt).